diff --git a/include/configs/mx6q_seco.h.orig b/include/configs/mx6q_seco.h
index 5bbd2e0..d2c9628 100644
--- a/include/configs/mx6q_seco.h.orig
+++ b/include/configs/mx6q_seco.h
@@ -248,7 +248,7 @@
 		"ipaddr=10.0.0.5\0"                                  \
                 "nfs_path=/targetfs \0"                                 \
                 "memory=mem=" ENV_SDRAM_SIZE "\0"			\
-                "bootdev=mmc dev 0; ext2load mmc 0:1\0"                 \
+                "bootdev=mmc dev 1; ext2load mmc 1:1\0"                 \
                 "root=root=/dev/mmcblk0p1\0"                            \
                 "option=rootwait rw fixrtc rootflags=barrier=1\0"       \
                 "setvideomode=setenv videomode video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24 setenv video1 video=mxcfb1:dev=ldb,${lvdsres1}\0"  \
